src/database.rs

```diff
@@ -1653,7 +1653,19 @@ fn resume_committing() {
     // Let us drop the DB to check if we're loading the uncommitted events
     // correctly.
     drop(db);
-    let mut db = Database::new(tmpdir.path()).unwrap();
+    let mut counter = 0;
+    let mut db = Database::new(tmpdir.path());
+
+    while db.is_err() {
```

Comment:
Can you add a quick comment (akin to the commit message) for why we're doing this.